In [1]:
import numpy as np
import pandas as pd
import sklearn

In [2]:
df_questions = pd.read_csv("../raw_data/deep_course_formulaire_question.csv")

In [3]:
#garde que les questions indiquées par Simon
liste=["F7664060CC52BC6F3D620BCEDC94A4B6","9A1158154DFA42CADDBD0694A4E9BDC8","AC1DD209CBCC5E5D1C6E28598E8CBBE8",
                "289DFF07669D7A23DE0EF88D2F7129E7","19F3CD308F1455B3FA09A282E0D496F4"]
df_questions_retenues=df_questions[df_questions["id"].isin(liste)]

In [4]:
#Suppression de colonnes inutiles
df_questions_retenues = df_questions_retenues.drop(labels = ["aide", "enonce_groupe", "type", "unite"], axis = 1)
df_questions_retenues.rename(columns = {'id': 'question_id'},inplace=True)
df_questions_retenues

enonce_site  \
46   Etes-vous soumis à l'obligation de réalisation...   
66   Avez-vous au moins l'un de ces équipements : h...   
184                  Utilisez-vous des nanomatériaux ?   
208       Avez-vous un conseiller en radioprotection ?   
306  Etes-vous considéré comme un consommateur gazo...   

                          question_id  
46   19F3CD308F1455B3FA09A282E0D496F4  
66   289DFF07669D7A23DE0EF88D2F7129E7  
184  9A1158154DFA42CADDBD0694A4E9BDC8  
208  AC1DD209CBCC5E5D1C6E28598E8CBBE8  
306  F7664060CC52BC6F3D620BCEDC94A4B6

In [5]:
df_reponses = pd.read_csv("../raw_data/deep_course_formulaire_reponse.csv")

In [6]:
#Base des réponses des clients pour les réponses retenues
df_reponses_retenues = df_reponses.loc[df_reponses['question_id'].isin(df_questions_retenues["question_id"])]

In [7]:
df_reponses_retenues = df_reponses_retenues.dropna(subset = ["valeur_actee"]).reset_index(drop=True)
df_reponses_retenues = df_reponses_retenues.drop(labels = ["non_concerne_acte", "commentaire_acte"], axis = 1)
df_reponses_retenues["valeur_actee"] = df_reponses_retenues["valeur_actee"].apply(lambda x : x.replace("-",""))  #on élimine les tirets
df_reponses_retenues["valeur_actee"] = df_reponses_retenues["valeur_actee"].str.upper() #On met en majuscules pour coincïder avec df_choix


In [8]:
df_reponses_retenues

node_id                      valeur_actee  \
0     2c90f5883f44af68013f564a1326558a  38457391C91D03A7DDDE63B07E5BBD91   
1     2c90f5883f44af68013f564a1326558a  40708C993C0699A2F01FDC779146E215   
2     2c90f5883fcf3281013fe7c2f0eb4dbe  2169F6D1F353C8873DD9E6BF3EC680CD   
3     2c90f5883fcf3281013fe7c2f0eb4dbe  78D3A8F9A8720821ED7970C1EA5BFB83   
4     2c90f5883fcf3281013fe7c2f0eb4dbe  4CA717FDBBA1508E012A38F3E30E6F7D   
5     2c90f5883fcf3281013fe7c2f0eb4dbe  40708C993C0699A2F01FDC779146E215   
6     2c90f5883fcf3281013fe7c2f0eb4dbe  E1DA2722BD6EB468459F304A2AA03604   
7     2c90f588404b48480140ce4d7d301013  78D3A8F9A8720821ED7970C1EA5BFB83   
8     2c90f588404b48480140ce4d7d301013  40708C993C0699A2F01FDC779146E215   
9     2c90f5884109d2c401410c40ae942d95  78D3A8F9A8720821ED7970C1EA5BFB83   
10    2c90f5884109d2c401410c40ae942d95  38457391C91D03A7DDDE63B07E5BBD91   
11    2c90f5884109d2c401410c40ae942d95  40708C993C0699A2F01FDC779146E215   
12    2c90f5884109d2c401410c4434a433a6  78D3A8F9A8720821ED7970C1EA5BFB83   
13    2c90f5884109d2c401410c4434a433a6  38457391C91D03A7DDDE63B07E5BBD91   
14    2c90f5884109d2c401410c4434a433a6  9ABD6E8782861517B838C81960960F65   
15    2c90f5884109d2c401410c44a3493440  78D3A8F9A8720821ED7970C1EA5BFB83   
16    2c90f5884109d2c401410c44a3493440  38457391C91D03A7DDDE63B07E5BBD91   
17    2c90f5884109d2c401410c44a3493440  9ABD6E8782861517B838C81960960F65   
18    2c90f588429b4a8a01429f3055457934  FCACABED7E6DA5A667FA723BB71EB327   
19    2c90f588429b4a8a01429f3055457934  38457391C91D03A7DDDE63B07E5BBD91   
20    2c90f588429b4a8a01429f30cc007a32  FCACABED7E6DA5A667FA723BB71EB327   
21    2c90f588429b4a8a01429f30cc007a32  38457391C91D03A7DDDE63B07E5BBD91   
22    2c90f588429b4a8a01429f30cc007a32  9ABD6E8782861517B838C81960960F65   
23    2c90f58845004b4601451c44694220d7  2169F6D1F353C8873DD9E6BF3EC680CD   
24    2c90f58845004b4601451c44694220d7  38457391C91D03A7DDDE63B07E5BBD91   
25    2c90f58845004b4601451c44694220d7  40708C993C0699A2F01FDC779146E215   
26    2c90f58845004b4601453c94255e139f  78D3A8F9A8720821ED7970C1EA5BFB83   
27    2c90f58845004b4601453c94255e139f  38457391C91D03A7DDDE63B07E5BBD91   
28    2c90f58845004b4601453c94255e139f  40708C993C0699A2F01FDC779146E215   
29    2c90f588456c31f50145acd97a197df7  78D3A8F9A8720821ED7970C1EA5BFB83   
...                                ...                               ...   
1497  ff80818171138064017134ddacb70d32  40708C993C0699A2F01FDC779146E215   
1498  ff80818171138064017134ddacb70d32  E1DA2722BD6EB468459F304A2AA03604   
1499  ff80818171138064017140422e46153a  78D3A8F9A8720821ED7970C1EA5BFB83   
1500  ff80818171138064017140422e46153a  38457391C91D03A7DDDE63B07E5BBD91   
1501  ff80818171138064017140422e46153a  9ABD6E8782861517B838C81960960F65   
1502  ff8081817113806401714042579716a2  FCACABED7E6DA5A667FA723BB71EB327   
1503  ff8081817113806401714042579716a2  38457391C91D03A7DDDE63B07E5BBD91   
1504  ff8081817113806401714042579716a2  9ABD6E8782861517B838C81960960F65   
1505  ff808181717a658e017181c0a2022261  78D3A8F9A8720821ED7970C1EA5BFB83   
1506  ff808181717a658e017181c0a2022261  38457391C91D03A7DDDE63B07E5BBD91   
1507  ff808181717a658e017181c0a2022261  9ABD6E8782861517B838C81960960F65   
1508  ff808181717a658e017181c124ca22c6  78D3A8F9A8720821ED7970C1EA5BFB83   
1509  ff808181717a658e017181c124ca22c6  38457391C91D03A7DDDE63B07E5BBD91   
1510  ff808181717a658e017181c124ca22c6  9ABD6E8782861517B838C81960960F65   
1511  ff808181725806fc017299554104555f  C4B831DA44BE304AF06AFFAFA22C413D   
1512  ff808181725806fc017299554104555f  38457391C91D03A7DDDE63B07E5BBD91   
1513  ff80818172c43c3a0172c675afe25945  78D3A8F9A8720821ED7970C1EA5BFB83   
1514  ff80818172c43c3a0172c675afe25945  38457391C91D03A7DDDE63B07E5BBD91   
1515  ff80818172c43c3a0172c675afe25945  9ABD6E8782861517B838C81960960F65   
1516  ff80818172c43c3a0172c675f04f5afc  78D3A8F9A8720821ED7970C1EA5BFB83   
1517  ff80818172c43c3a0172c675f04f5afc  38457391C91D03A7DDDE63B07E5BBD91   
151

In [10]:
#Choix possible des réponses pour les questions retenues
df_choix = pd.read_csv("../raw_data/deep_course_formulaire_question_choix.csv", low_memory = False)
df_choix = df_choix.loc[df_choix['question_id'].isin(df_questions_retenues["question_id"])]
df_choix

choix_order                          choix_id  \
70             0  2169F6D1F353C8873DD9E6BF3EC680CD   
71             1  C4B831DA44BE304AF06AFFAFA22C413D   
106            0  78D3A8F9A8720821ED7970C1EA5BFB83   
107            1  FCACABED7E6DA5A667FA723BB71EB327   
326            0  4CA717FDBBA1508E012A38F3E30E6F7D   
327            1  38457391C91D03A7DDDE63B07E5BBD91   
372            0  40708C993C0699A2F01FDC779146E215   
373            1  9ABD6E8782861517B838C81960960F65   
555            0  C5EA8003E8169156A99F97C7D73055AF   
556            1  E1DA2722BD6EB468459F304A2AA03604   

                          question_id  
70   19F3CD308F1455B3FA09A282E0D496F4  
71   19F3CD308F1455B3FA09A282E0D496F4  
106  289DFF07669D7A23DE0EF88D2F7129E7  
107  289DFF07669D7A23DE0EF88D2F7129E7  
326  9A1158154DFA42CADDBD0694A4E9BDC8  
327  9A1158154DFA42CADDBD0694A4E9BDC8  
372  AC1DD209CBCC5E5D1C6E28598E8CBBE8  
373  AC1DD209CBCC5E5D1C6E28598E8CBBE8  
555  F7664060CC52BC6F3D620BCEDC94A4B6  
556  F7664060CC52BC6F3D620BCEDC94A4B6

In [11]:
#réponses (avec 1 ou 0) des clients aux questions 
df_rep = pd.merge(df_reponses_retenues, df_choix, how='left', left_on=['valeur_actee'], right_on=['choix_id'])
df_rep

node_id                      valeur_actee  \
0     2c90f5883f44af68013f564a1326558a  38457391C91D03A7DDDE63B07E5BBD91   
1     2c90f5883f44af68013f564a1326558a  40708C993C0699A2F01FDC779146E215   
2     2c90f5883fcf3281013fe7c2f0eb4dbe  2169F6D1F353C8873DD9E6BF3EC680CD   
3     2c90f5883fcf3281013fe7c2f0eb4dbe  78D3A8F9A8720821ED7970C1EA5BFB83   
4     2c90f5883fcf3281013fe7c2f0eb4dbe  4CA717FDBBA1508E012A38F3E30E6F7D   
5     2c90f5883fcf3281013fe7c2f0eb4dbe  40708C993C0699A2F01FDC779146E215   
6     2c90f5883fcf3281013fe7c2f0eb4dbe  E1DA2722BD6EB468459F304A2AA03604   
7     2c90f588404b48480140ce4d7d301013  78D3A8F9A8720821ED7970C1EA5BFB83   
8     2c90f588404b48480140ce4d7d301013  40708C993C0699A2F01FDC779146E215   
9     2c90f5884109d2c401410c40ae942d95  78D3A8F9A8720821ED7970C1EA5BFB83   
10    2c90f5884109d2c401410c40ae942d95  38457391C91D03A7DDDE63B07E5BBD91   
11    2c90f5884109d2c401410c40ae942d95  40708C993C0699A2F01FDC779146E215   
12    2c90f5884109d2c401410c4434a433a6  78D3A8F9A8720821ED7970C1EA5BFB83   
13    2c90f5884109d2c401410c4434a433a6  38457391C91D03A7DDDE63B07E5BBD91   
14    2c90f5884109d2c401410c4434a433a6  9ABD6E8782861517B838C81960960F65   
15    2c90f5884109d2c401410c44a3493440  78D3A8F9A8720821ED7970C1EA5BFB83   
16    2c90f5884109d2c401410c44a3493440  38457391C91D03A7DDDE63B07E5BBD91   
17    2c90f5884109d2c401410c44a3493440  9ABD6E8782861517B838C81960960F65   
18    2c90f588429b4a8a01429f3055457934  FCACABED7E6DA5A667FA723BB71EB327   
19    2c90f588429b4a8a01429f3055457934  38457391C91D03A7DDDE63B07E5BBD91   
20    2c90f588429b4a8a01429f30cc007a32  FCACABED7E6DA5A667FA723BB71EB327   
21    2c90f588429b4a8a01429f30cc007a32  38457391C91D03A7DDDE63B07E5BBD91   
22    2c90f588429b4a8a01429f30cc007a32  9ABD6E8782861517B838C81960960F65   
23    2c90f58845004b4601451c44694220d7  2169F6D1F353C8873DD9E6BF3EC680CD   
24    2c90f58845004b4601451c44694220d7  38457391C91D03A7DDDE63B07E5BBD91   
25    2c90f58845004b4601451c44694220d7  40708C993C0699A2F01FDC779146E215   
26    2c90f58845004b4601453c94255e139f  78D3A8F9A8720821ED7970C1EA5BFB83   
27    2c90f58845004b4601453c94255e139f  38457391C91D03A7DDDE63B07E5BBD91   
28    2c90f58845004b4601453c94255e139f  40708C993C0699A2F01FDC779146E215   
29    2c90f588456c31f50145acd97a197df7  78D3A8F9A8720821ED7970C1EA5BFB83   
...                                ...                               ...   
1497  ff80818171138064017134ddacb70d32  40708C993C0699A2F01FDC779146E215   
1498  ff80818171138064017134ddacb70d32  E1DA2722BD6EB468459F304A2AA03604   
1499  ff80818171138064017140422e46153a  78D3A8F9A8720821ED7970C1EA5BFB83   
1500  ff80818171138064017140422e46153a  38457391C91D03A7DDDE63B07E5BBD91   
1501  ff80818171138064017140422e46153a  9ABD6E8782861517B838C81960960F65   
1502  ff8081817113806401714042579716a2  FCACABED7E6DA5A667FA723BB71EB327   
1503  ff8081817113806401714042579716a2  38457391C91D03A7DDDE63B07E5BBD91   
1504  ff8081817113806401714042579716a2  9ABD6E8782861517B838C81960960F65   
1505  ff808181717a658e017181c0a2022261  78D3A8F9A8720821ED7970C1EA5BFB83   
1506  ff808181717a658e017181c0a2022261  38457391C91D03A7DDDE63B07E5BBD91   
1507  ff808181717a658e017181c0a2022261  9ABD6E8782861517B838C81960960F65   
1508  ff808181717a658e017181c124ca22c6  78D3A8F9A8720821ED7970C1EA5BFB83   
1509  ff808181717a658e017181c124ca22c6  38457391C91D03A7DDDE63B07E5BBD91   
1510  ff808181717a658e017181c124ca22c6  9ABD6E8782861517B838C81960960F65   
1511  ff808181725806fc017299554104555f  C4B831DA44BE304AF06AFFAFA22C413D   
1512  ff808181725806fc017299554104555f  38457391C91D03A7DDDE63B07E5BBD91   
1513  ff80818172c43c3a0172c675afe25945  78D3A8F9A8720821ED7970C1EA5BFB83   
1514  ff80818172c43c3a0172c675afe25945  38457391C91D03A7DDDE63B07E5BBD91   
1515  ff80818172c43c3a0172c675afe25945  9ABD6E8782861517B838C81960960F65   
1516  ff80818172c43c3a0172c675f04f5afc  78D3A8F9A8720821ED7970C1EA5BFB83   
1517  ff80818172c43c3a0172c675f04f5afc  38457391C91D03A7DDDE63B07E5BBD91   
151

In [12]:
df_rep = df_rep.drop(labels = ["valeur_actee", "question_id_x","choix_id"], axis = 1)

In [13]:
#Garde que les réponses "Oui" (du moins les 1) 
#Autre base mais avec les 0
indexNames1 = df_rep[ df_rep['choix_order'] == 0 ].index
indexNames0 = df_rep[ df_rep['choix_order'] == 1 ].index
df_rep0 = df_rep.drop(indexNames0)
df_rep.drop(indexNames1 , inplace=True)
df_rep.rename(columns = {'question_id_y': 'question_id'},inplace=True)
df_rep0.rename(columns = {'question_id_y': 'question_id'},inplace=True)
df_rep

node_id  choix_order  \
0     2c90f5883f44af68013f564a1326558a            1   
6     2c90f5883fcf3281013fe7c2f0eb4dbe            1   
10    2c90f5884109d2c401410c40ae942d95            1   
13    2c90f5884109d2c401410c4434a433a6            1   
14    2c90f5884109d2c401410c4434a433a6            1   
16    2c90f5884109d2c401410c44a3493440            1   
17    2c90f5884109d2c401410c44a3493440            1   
18    2c90f588429b4a8a01429f3055457934            1   
19    2c90f588429b4a8a01429f3055457934            1   
20    2c90f588429b4a8a01429f30cc007a32            1   
21    2c90f588429b4a8a01429f30cc007a32            1   
22    2c90f588429b4a8a01429f30cc007a32            1   
24    2c90f58845004b4601451c44694220d7            1   
27    2c90f58845004b4601453c94255e139f            1   
30    2c90f588456c31f50145acd97a197df7            1   
32    2c90f588456c31f50145acdd5b7705e7            1   
33    2c90f588456c31f50145acdd5b7705e7            1   
35    2c90f588456c31f50145acddb7dd0605            1   
36    2c90f588456c31f50145acddb7dd0605            1   
38    2c90f5884b96e139014ba78902d02598            1   
40    2c90f5884c0ad6db014c312f658e386f            1   
42    2c90f5884c600d3f014c98394833335d            1   
45    2c90f5884c600d3f014c9847885452d8            1   
49    2c90f5884c600d3f014c98486bce540c            1   
51    2c90f5884c600d3f014c9848b08d54dc            1   
52    2c90f5884c600d3f014c9848b08d54dc            1   
53    2c90f5884c600d3f014c9848faa1568a            1   
55    2c90f5884c600d3f014c9848faa1568a            1   
56    2c90f5884c600d3f014c9848faa1568a            1   
58    2c90f5884c600d3f014c98493f43585f            1   
...                                ...          ...   
1483  ff80818170122875017014d8f6886cd0            1   
1485  ff808181706ea80601707bfd5c675316            1   
1486  ff808181706ea80601707bfd5c675316            1   
1488  ff80818170c6ae290170d4376d953371            1   
1489  ff80818170c6ae290170d4376d953371            1   
1492  ff80818170c6ae290170e818cada7550            1   
1493  ff80818170c6ae290170e818cada7550            1   
1496  ff80818171138064017134ddacb70d32            1   
1498  ff80818171138064017134ddacb70d32            1   
1500  ff80818171138064017140422e46153a            1   
1501  ff80818171138064017140422e46153a            1   
1502  ff8081817113806401714042579716a2            1   
1503  ff8081817113806401714042579716a2            1   
1504  ff8081817113806401714042579716a2            1   
1506  ff808181717a658e017181c0a2022261            1   
1507  ff808181717a658e017181c0a2022261            1   
1509  ff808181717a658e017181c124ca22c6            1   
1510  ff808181717a658e017181c124ca22c6            1   
1511  ff808181725806fc017299554104555f            1   
1512  ff808181725806fc017299554104555f            1   
1514  ff80818172c43c3a0172c675afe25945            1   
1515  ff80818172c43c3a0172c675afe25945            1   
1517  ff80818172c43c3a0172c675f04f5afc            1   
1518  ff80818172c43c3a0172c675f04f5afc            1   
1520  ff80818172c43c3a0172c7a3d28c62e5            1   
1521  ff80818172c43c3a0172c7a3d28c62e5            1   
1522  ff80818173787da801739e854673578c            1   
1524  ff80818173787da801739e854673578c            1   
1525  ff80818173787da801739e854673578c            1   
1526  ff80818173787da801739e854673578c            1   

                           question_id  
0     9A1158154DFA42CADDBD0694A4E9BDC8  
6     F7664060CC52BC6F3D620BCEDC94A4B6  
10    9A1158154DFA42CADDBD0694A4E9BDC8  
13    9A1158154DFA42CADDBD0694A4E9BDC8  
14    AC1DD209CBCC5E5D1C6E28598E8CBBE8  
16    9A1158154DFA42CADDBD0694A4E9BDC8  
17    AC1DD209CBCC5E5D1C6E28598E8CBBE8  
18    289DFF07669D7A23DE0EF88D2F7129E7  
19    9A1158154DFA42CADDBD0694A4E9BDC8  
20    289DFF07669D7A23DE0EF88D2F7129E7  
21    9A1158154DFA42CADDBD0694A4E9BDC8  
22    AC1DD209CBCC5E5D1C6E28598E8CBBE8  
24    9A1158154DFA42CADDBD0694A4E9BDC8  
27    9A1158154DFA42CADDBD0694A4E9BDC8  
30   

In [18]:
#Mets les clients dans une liste en fonction de leurs réponses aux questions retenues
list_clients_gazo = df_rep[ df_rep['question_id'] == "F7664060CC52BC6F3D620BCEDC94A4B6" ]
list_clients_gazo = list(list_clients_gazo["node_id"])

list_clients_nanomateriaux = df_rep[ df_rep['question_id'] == "9A1158154DFA42CADDBD0694A4E9BDC8" ]
list_clients_nanomateriaux = list(list_clients_nanomateriaux["node_id"])

list_clients_radioprotection = df_rep[ df_rep['question_id'] == "AC1DD209CBCC5E5D1C6E28598E8CBBE8" ]
list_clients_radioprotection = list(list_clients_radioprotection["node_id"])

list_clients_equipements = df_rep[ df_rep['question_id'] == "289DFF07669D7A23DE0EF88D2F7129E7" ]
list_clients_equipements = list(list_clients_equipements["node_id"])

list_clients_GES = df_rep[ df_rep['question_id'] == "19F3CD308F1455B3FA09A282E0D496F4" ]
list_clients_GES = list(list_clients_GES["node_id"])


In [15]:
#Pareil mais pour les réponses 0
list_clients_gazo0 = df_rep0[ df_rep0['question_id'] == "F7664060CC52BC6F3D620BCEDC94A4B6" ]
list_clients_gazo0 = list(list_clients_gazo0["node_id"])

list_clients_nanomateriaux0 = df_rep0[ df_rep0['question_id'] == "9A1158154DFA42CADDBD0694A4E9BDC8" ]
list_clients_nanomateriaux0 = list(list_clients_nanomateriaux0["node_id"])

list_clients_radioprotection0 = df_rep0[ df_rep0['question_id'] == "AC1DD209CBCC5E5D1C6E28598E8CBBE8" ]
list_clients_radioprotection0 = list(list_clients_radioprotection0["node_id"])

list_clients_equipements0 = df_rep0[ df_rep0['question_id'] == "289DFF07669D7A23DE0EF88D2F7129E7" ]
list_clients_equipements0 = list(list_clients_equipements0["node_id"])

list_clients_GES0 = df_rep0[ df_rep0['question_id'] == "19F3CD308F1455B3FA09A282E0D496F4" ]
list_clients_GES0 = list(list_clients_GES0["node_id"])


In [16]:
#Création d'une DB avec juste les clients et les textes qui leur sont attribués (groupe ID)
df_text_versions_node = pd.read_csv("../raw_data/deep_course_cf_text_version_nodes.csv", low_memory = False)
df_text_versions = pd.read_csv("../raw_data/deep_course_cf_text_versions.csv", low_memory = False)
df_text_versions = df_text_versions.drop(labels = ["txt_name", "theme_id","type_id"], axis = 1)
df_text = pd.merge(df_text_versions_node, df_text_versions, how='left', left_on=['txt_version_surrogate_uuid'], right_on=['surrogate_uuid'])
df_text = df_text.drop(labels = ["txt_version_surrogate_uuid", "surrogate_uuid"], axis=1)
df_text

txt_node_id  group_id
0       2c90f5883f44af68013f564a1326558a   TXA9448
1       2c90f5883f44af68013f564a1326558a   TXA9312
2       2c90f5883f44af68013f564a1326558a   TXA9229
3       2c90f5883f44af68013f564a1326558a   TXA9212
4       2c90f5883f44af68013f564a1326558a   TXA9136
5       2c90f5883f44af68013f564a1326558a   TXA9126
6       2c90f5883f44af68013f564a1326558a   TXA9107
7       2c90f5883f44af68013f564a1326558a   TXA9068
8       2c90f5883f44af68013f564a1326558a   TXA8883
9       2c90f5883f44af68013f564a1326558a   TXA8876
10      2c90f5883f44af68013f564a1326558a   TXA8867
11      2c90f5883f44af68013f564a1326558a   TXA8833
12      2c90f5883f44af68013f564a1326558a   TXA8672
13      2c90f5883f44af68013f564a1326558a   TXA8592
14      2c90f5883f44af68013f564a1326558a   TXA8360
15      2c90f5883f44af68013f564a1326558a   TXA8189
16      2c90f5883f44af68013f564a1326558a   TXA7936
17      2c90f5883f44af68013f564a1326558a   TXA7935
18      2c90f5883f44af68013f564a1326558a   TXA7934
19      2c90f5883f44af68013f564a1326558a   TXA7932
20      2c90f5883f44af68013f564a1326558a   TXA7703
21      2c90f5883f44af68013f564a1326558a   TXA7686
22      2c90f5883f44af68013f564a1326558a   TXA7685
23      2c90f5883f44af68013f564a1326558a   TXA7657
24      2c90f5883f44af68013f564a1326558a   TXA7656
25      2c90f5883f44af68013f564a1326558a   TXA7525
26      2c90f5883f44af68013f564a1326558a   TXA7513
27      2c90f5883f44af68013f564a1326558a   TXA7408
28      2c90f5883f44af68013f564a1326558a   TXA7407
29      2c90f5883f44af68013f564a1326558a   TXA7281
...                                  ...       ...
329912  ff80818173787da801739e854673578c  TXA10519
329913  ff80818173787da801739e854673578c   TXA6431
329914  ff80818173787da801739e854673578c   TXA6430
329915  ff80818173787da801739e854673578c   TXA5707
329916  ff80818173787da801739e854673578c  TXA11507
329917  ff80818173787da801739e854673578c  TXA10603
329918  ff80818173787da801739e854673578c   TXA3400
329919  ff80818173787da801739e854673578c   TXA9983
329920  ff80818173787da801739e854673578c   TXA5125
329921  ff80818173787da801739e854673578c   TXA3372
329922  ff80818173787da801739e854673578c  TXA10185
329923  ff80818173787da801739e854673578c   TXA6665
329924  ff80818173787da801739e854673578c   TXA9789
329925  ff80818173787da801739e854673578c   TXA9788
329926  ff80818173787da801739e854673578c   TXA6004
329927  ff80818173787da801739e854673578c   TXA3699
329928  ff80818173787da801739e854673578c   TXA6230
329929  ff80818173787da801739e854673578c   TXA3417
329930  ff80818173787da801739e854673578c   TXA8062
329931  ff80818173787da801739e854673578c   TXA3704
329932  ff80818173787da801739e854673578c   TXA7601
329933  ff80818173787da801739e854673578c   TXA7001
329934  ff80818173787da801739e854673578c  TXA13049
329935  ff80818173787da801739e854673578c  TXA13050
329936  ff80818173787da801739e854673578c  TXA12853
329937  ff80818173787da801739e854673578c   TXA9803
329938  ff80818173787da801739e854673578c   TXA8985
329939  ff80818173787da801739e854673578c   TXA8984
329940  ff80818173787da801739e854673578c   TXA9060
329941  ff80818173787da801739e854673578c   TXA3271

[329942 rows x 2 columns]

In [52]:
#définission d'une fonction qui va prendre en compter l'occurrence d'un texte dans un tableau

def count(df,txt):
    c=0
    for i in np.arange(df.shape[0]) :
        if df.iloc[i,1]==txt :
            c=c+1
    return c

### Vérification pour la réponse "oui" à la question des nanomatériaux

In [19]:
#creation de la base qui lie les clients qui ont répondu oui pour les nanomatériaux et les textes qui leur sont attribués
df_txt_client_nanomateriaux=df_text[df_text["txt_node_id"].isin(list_clients_nanomateriaux)]

In [20]:
df_txt_client_nanomateriaux.describe()

txt_node_id group_id
count                             231317   226373
unique                               392     3375
top     d37394c52f494d90012f4e0d1f4426fc  TXA3418
freq                                1578      384

Il y a 392 clients dans la db qui ont répondu oui à la question sur les nanomatériaux, donc normalement, on devrait voir apparaître 392 fois les textes indiqués par Simon

In [21]:
df_txt_client_nanomateriaux['group_id'].value_counts()[0:10]

TXA3418     384
TXA6461     384
TXA9136     382
TXA6459     382
TXA6472     381
TXA6475     381
TXA6462     381
TXA10136    378
TXA6426     376
TXA6471     376
Name: group_id, dtype: int64

In [55]:
print(count(df_txt_client_nanomateriaux,"TXA9133"),count(df_txt_client_nanomateriaux,"TXA8556"),
      count(df_txt_client_nanomateriaux,"TXA9266"))

55 41 47


Problématique, car en revenant aux textes les plus présents parmi les clients qui ont répondu oui à la question sur les nanomatériaux, ils n'apparaissent même pas dans les top 25

In [22]:
#creation de la base qui lie les clients qui ont répondu "0" pour les nanomatériaux et les textes qui leur sont attribués
df_txt_client_nanomateriaux0=df_text[df_text["txt_node_id"].isin(list_clients_nanomateriaux0)]

In [24]:
df_txt_client_nanomateriaux0.describe()

txt_node_id group_id
count                              32724    31835
unique                                42     2619
top     4028807d16245e4d0116249e4259205e  TXA6471
freq                                1641       42

In [26]:
df_txt_client_nanomateriaux0['group_id'].value_counts()

TXA6471     42
TXA6472     42
TXA11510    42
TXA3629     42
TXA6461     42
TXA6426     42
TXA11367    42
TXA6475     42
TXA6462     42
TXA9136     42
TXA3418     42
TXA5924     42
TXA6459     42
TXA10136    41
TXA10130    41
TXA6468     41
TXA6882     40
TXA3323     40
TXA5251     40
TXA6886     40
TXA6888     40
TXA3248     40
TXA5707     40
TXA9132     40
TXA6759     40
TXA3372     40
TXA7403     40
TXA5876     40
TXA8580     40
TXA6892     40
            ..
TXA3371      1
TXA6910      1
TXA6241      1
TXA6863      1
TXA8949      1
TXA6889      1
TXA11460     1
TXA11379     1
TXA11416     1
TXA8501      1
TXA5951      1
TXA7134      1
TXA8872      1
TXA11404     1
TXA9672      1
TXA3345      1
TXA11254     1
TXA3583      1
TXA11708     1
TXA11504     1
TXA4705      1
TXA5339      1
TXA11889     1
TXA8761      1
TXA5793      1
TXA11310     1
TXA3691      1
TXA8760      1
TXA10741     1
TXA9403      1
Name: group_id, Length: 2619, dtype: int64

In [56]:
print(count(df_txt_client_nanomateriaux0,"TXA9133"),count(df_txt_client_nanomateriaux0,"TXA8556"),
      count(df_txt_client_nanomateriaux0,"TXA9266"))

28 28 25


### Vérification pour la réponse "oui" à la question gazo-intensif

In [28]:
df_txt_client_gazo=df_text[df_text["txt_node_id"].isin(list_clients_gazo)]
df_txt_client_gazo.describe()

txt_node_id  group_id
count                              50291     49237
unique                                64      2687
top     d37394c52f494d90012f4e0d1f4426fc  TXA11510
freq                                1578        64

In [57]:
df_txt_client_gazo['group_id'].value_counts()[0:5]

TXA11510    64
TXA10842    64
TXA10500    63
TXA6426     63
TXA3372     63
Name: group_id, dtype: int64

In [58]:
print(count(df_txt_client_gazo,"TXA9547"),count(df_txt_client_gazo,"TXA10065"))

1 3


In [60]:
df_txt_client_gazo0=df_text[df_text["txt_node_id"].isin(list_clients_gazo0)]

In [61]:
print(count(df_txt_client_gazo0,"TXA9547"),count(df_txt_client_gazo0,"TXA10065"))

6 6


### Vérification pour la réponse "oui" à la question radio-protection

In [62]:
df_txt_client_radioprotection=df_text[df_text["txt_node_id"].isin(list_clients_radioprotection)]
df_txt_client_radioprotection.describe()

txt_node_id group_id
count                             169567   166957
unique                               293     3255
top     d37394c52f494d90012f4e0d1f4426fc  TXA3418
freq                                1578      286

In [64]:
df_txt_client_radioprotection['group_id'].value_counts()[0:5]

TXA3418    286
TXA6461    286
TXA6472    284
TXA6475    283
TXA6459    283
Name: group_id, dtype: int64

In [68]:
print(count(df_txt_client_radioprotection,"TXA6921"),count(df_txt_client_radioprotection,"TXA12264"))

111 65


In [66]:
df_txt_client_radioprotection0=df_text[df_text["txt_node_id"].isin(list_clients_radioprotection0)]
df_txt_client_radioprotection0.describe()

txt_node_id group_id
count                              92570    89247
unique                               126     3035
top     4028807d16245e4d0116249e4259205e  TXA3418
freq                                1641      124

In [69]:
print(count(df_txt_client_radioprotection0,"TXA6921"),count(df_txt_client_radioprotection0,"TXA12264"))

104 102
